src/models/users.rs

```diff
@@ -89,6 +89,22 @@ impl From<DBUser> for User {
     }
 }
 
+impl User {
+    pub fn validate_can_interact_with_oauth_client(
+        &self,
+        client_owner_id: UserId,
+    ) -> Result<(), crate::routes::ApiError> {
+        if self.role.is_mod() || self.id == client_owner_id {
+            Ok(())
+        } else {
+            Err(crate::routes::ApiError::CustomAuthentication(
+                "You don't have sufficient permissions to interact with this OAuth application"
+                    .to_string(),
+            ))
+        }
+    }
+}
```

Comment:
weird place to put this method. imo this should go in the auth checks file or be a private func in the module that uses it

Reply:
The method is on the struct that has the data it operates on. This makes it easier to discover that the method exists when you need it (helping to avoid duplication) and makes it less work to call.

The other bit of info it needs is the owner of the client - so another good place I could see this living is on the OAuthClient model. That might be better, might do that. Then it would also (kind of) be in the same conceptual module as where it's used.

Reply:
Moved into `auth/checks.rs` AND is accessible through OAuthClient, thanks to ✨Rust traits✨